Recruit-Training/Basic-Recruit-Training/pointers/main.cpp

```diff
@@ -19,33 +20,36 @@ int main()
     int *p = &x;
 
     //TODO in the line below predict what what is going to be output
-    cout << "Predicted value of p: " /* << PLACE YOUR PREDICTION HERE*/ << endl;
+    cout << "Predicted value of p: "  << "address of x " << &x << endl;
     cout << "Actual value of p: " << p << endl;
-    cout << "Predicted value &x: " /* << PLACE YOUR PREDICTION HERE*/ << endl;
+    cout << "Predicted value &x: " << "address of x" << &x << endl;
     cout << "Actual value &x: " << &x << endl;
-    cout << "Predicted value of *p: " /* << PLACE YOUR PREDICTION HERE*/ << endl;
+    cout << "Predicted value of *p: " << 3 << endl;
     cout << "Actual value of *p: " << *p << endl;
     
     foo(p, x);
     
-    cout << "Predicted value of *p: " /* << PLACE YOUR PREDICTION HERE*/ << endl;
+    cout << "Predicted value of *p: " << 42 << endl;
     cout << "Actual value of *p: " << *p << endl;
-    cout << "Predicted value of x: " /* << PLACE YOUR PREDICTION HERE*/ << endl;
+    cout << "Predicted value of x: " << 42 << endl;
     cout << "Actual value of x: " << x << endl;
     foo(p, y);
 
-    cout << "Predicted value of *p: " /* << PLACE YOUR PREDICTION HERE*/ << endl;
+    cout << "Predicted value of *p: " << 42 << endl;
     cout << "Actual value of *p: " << *p << endl;
-    cout << "Predicted value of y: " /* << PLACE YOUR PREDICTION HERE*/ << endl;
+    cout << "Predicted value of y: " << 4 << endl;
     cout << "Actual value of y: " << y << endl;
-
-
-    //start writing MySwap here
     
```

Comment:
From the instructions at the top of the file, please make sure you call `MySwap()` in `main()` and pass in the pointer `p` and another pointer that points to `y`.

Reply:
`You will pass two variables in as argument, one will be the pointer p. The other one will be an integer pointer you make that points to the variable y.
In the function MySwap you will swap the values of the two arguments so the values are changed inside main()`. Then you can `std::cout` the values to check.